src/storage/lock_manager/deadlock.rs

```diff
@@ -474,12 +510,17 @@ impl Detector {
                     key_hash,
                     ..
                 } = req.get_entry();
+
+                let mut inner = inner.borrow_mut();
+                if *txn > inner.max_ts {
```

Comment:
Could we release the borrow_mut after changing the `max_ts`?

Reply:
Maybe define a function to change the max_ts if needed for the `inner` struct?

Reply:
Do you mean
```rust
                inner.borrow_mut().update_max_ts_if_needed(*txn);
                let inner = inner.borrow();
                let mut detect_table = inner.detect_table.borrow_mut();
```
instead of 
```rust
                let mut inner = inner.borrow_mut();
                inner.update_max_ts_if_needed(*txn);
                let mut detect_table = inner.detect_table.borrow_mut();
```

Reply:
@AndreMouche